processor/src/main/kotlin/com/syouth/kmapper/processor/base/Utils.kt

```diff
@@ -41,7 +41,10 @@ internal fun KSFunctionDeclaration.getMappingInformation(): MapperInformation {
     val from = parameters[0].type.resolve()
     val to = returnType!!.resolve()
     if (from.declaration !is KSClassDeclaration || to.declaration !is KSClassDeclaration) throw IllegalStateException("Mapping should be from class to class.")
-    if (!from.isDataClass() || !to.isDataClass()) throw IllegalStateException("Mapping is only supported for data classes")
+    if (!(from.isDataClass() && to.isDataClass()) &&
+        !(from.isDataClass() && !to.isDataClass()) &&
+        !(!from.isDataClass() && to.isDataClass()))
+        throw IllegalStateException("Mapping is only supported for data classes")
```

Comment:
This is not informative exception in case of the edits above.

Reply:
I've changed the check and the exception message to reflect the current status for now.